### PR TITLE
feat(dropdown):Add tests for dropdown

### PR DESCRIPTION
**Pull request by udhadpa**
Hello All,

Please review my tests for dropdown (Label, Button and Icon).
It covers both desktop and mobile tests.

Do let me know your feedback/suggestions.
@mhomolak @JasonLantz @StommePoes @wyseguyonline @umahaea 
Thanks, 
Payal

### CODE REVIEW

File: src/main/java/compounds/css/elements.css [hunk 1]
```diff
@@ -0,0 +1,2727 @@
+/*! normalize.css v5.0.0 | MIT License | github.com/necolas/normalize.css */
```
Comment (umahaea) [inline src/main/java/compounds/css/elements.css]: We got to remove this.

File: src/main/java/compounds/jsfiles/inputs/inputs.js [hunk 2]
```diff
@@ -10,6 +10,7 @@ function init() {
 				id: 'a',
 				label: 'FirstName',
 				inputType: 'default',
+				fancy: false,
```
Comment (umahaea) [inline src/main/java/compounds/jsfiles/inputs/inputs.js]: jus a question. Why do we have to change this value?

File: src/main/resources/environment.properties [hunk 3]
```diff
@@ -20,10 +20,10 @@ appiumVer=1.6.4
 #Desktop:-
 #        |platform    |sauceBrowser |version |
 #        |OS X 10.11  |chrome       |57.0    |
-#        |OS X 10.11  |firefox      |52.0    |
+#        |OS X 10.11  |firefox      |51.0    |
```
Comment (umahaea) [inline src/main/resources/environment.properties]: Lets set this to 52.0

File: src/main/shell_scripts/components.sh [hunk 4]
```diff
@@ -26,9 +26,9 @@ npm install &>/dev/null
 npm run build &>/dev/null
 cp ~/build/Pearson-Higher-Ed/ux-test-platform/compounds/build/dist.compounds.js ~/build/Pearson-Higher-Ed/ux-test-platform/src/main/java/compounds/jsfiles/
 cp ~/build/Pearson-Higher-Ed/ux-test-platform/compounds/build/dev.compounds.js ~/build/Pearson-Higher-Ed/ux-test-platform/src/main/java/compounds/jsfiles/
-cp -R ~/build/Pearson-Higher-Ed/ux-test-platform/compounds/node_modules/pearson-elements/dist/fonts ~/build/Pearson-Higher-Ed/ux-test-platform/
-cp -R ~/build/Pearson-Higher-Ed/ux-test-platform/compounds/node_modules/pearson-elements/dist/icons ~/build/Pearson-Higher-Ed/ux-test-platform/
-cp ~/build/Pearson-Higher-Ed/ux-test-platform/compounds/node_modules/pearson-elements/dist/css/elements.css ~/build/Pearson-Higher-Ed/ux-test-platform/src/main/java/compounds/css/
+#cp -R ~/build/Pearson-Higher-Ed/ux-test-platform/compounds/node_modules/pearson-elements/dist/fonts ~/build/Pearson-Higher-Ed/ux-test-platform/
```
Comment (umahaea) [inline src/main/shell_scripts/components.sh]: We do need to have these values enabled.

File: src/test/java/compoundsTests/DropdownTest.java [hunk 5]
```diff
@@ -0,0 +1,936 @@
...
+                {dropdownPgObj.mobileHeader, "line-height", new String[]{"22px"}},
+                {dropdownPgObj.mobileHeader, "border-bottom-color", new String[]{commonUtils.hex2Rgb("#e9e9e9"), commonUtils.hex2RgbWithoutTransparency("#e9e9e9")}},
+                {dropdownPgObj.mobileHeader, "border-bottom-style", new String[]{"solid"}},
+                {dropdownPgObj.mobileHeader, "border-bottom-width", new String[]{"1px"}},
+                {dropdownPgObj.mobileHeader, "margin-bottom", new String[]{"12px"}},
+                {dropdownPgObj.closeBtn, "margin-left", new String[]{"12px"}}
+        };
+    }
+
+    @Test(testName = "Label Dropdown Header Mobile Test", dataProvider = "Dropdown Header Mobile Test Data", groups = "mobile-regression")
```
Comment (umahaea) [inline src/test/java/compoundsTests/DropdownTest.java]: We usually have a format of writing our testName as "Mobile: XXXXXXX", to be consistent with other tests. Can you rename it?

File: src/test/java/compoundsTests/DropdownTest.java [hunk 6]
```diff
@@ -0,0 +1,936 @@
...
+        isCSSProperty = commonUtils.assertCSSProperties(cssPropertyType, cssProperty, expectedCSSValue);
+        if (!isCSSProperty) {
+            log.info("Compounds-> '" + cssPropertyType + "' :of Header for button Dropdown is not as per the spec, actual: " + cssProperty);
+        }
+        Assert.assertTrue(isCSSProperty);
+    }
+
+    @Test(testName = "Dropdown Options Mobile Test", dataProvider = "Dropdown Options Test Data", groups = "mobile-regression")
+    private void optionsLabelDropdownMobileTest(String dropdownType, String type, By elem, By trigger, String expPaddingLeft, String expPaddingRight, String expPaddingTop, String expPaddingBtm, String expFontSize, String expLineHt) throws InterruptedException, IOException {
+        if (mobileDevice.equals("iPhone 6s Plus Simulator") || mobileDevice.equals("iPhone 7 Plus Simulator")) {
```
Comment (umahaea) [inline src/test/java/compoundsTests/DropdownTest.java]: For which responsive size does this test work? The 'if' condition here will skip both these devices.

File: src/test/java/compoundsTests/DropdownTest.java [hunk 7]
```diff
@@ -0,0 +1,936 @@
...
+        return path;
+    }
+
+    private void focusOnOptions(By option) {
+        WebElement elem = driver.findElement(option);
+        new Actions(driver).moveToElement(elem).perform();
+    }
+
+    //Override a file contents with new config
+    public void changeConfig(String jsFilePath, String getTestConfig) throws IOException, InterruptedException {
```
Comment (umahaea) [inline src/test/java/compoundsTests/DropdownTest.java]: We do have a commonUtils method to change the config. Is this a duplicate?

File: src/test/java/compoundsTests/DropdownTest.java [hunk 8]
```diff
@@ -0,0 +1,936 @@
...
+        if (!dropdownType.equals("icon")) {
+            String[] propsPropertiesList = new String[]{"presentationType", dropdownType, "presentationText", dropdownType, "list", "['Thing one', 'Thing two']", "mobileTitle", "Mobile title"};
+            testConfig = buildJSONObjectDetailConfig(detailsPropertiesList, propsPropertiesList);
+        } else {
+            String[] propsPropertiesList = new String[]{"presentationType", "icon", "list", "['Pearson', 'Design','divider', 'Accelerator!!']", "mobileTitle", "Mobile title"};
+            testConfig = buildJSONObjectDetailConfig(detailsPropertiesList, propsPropertiesList);
+        }
+        changeConfig(dropdownJSFilePath, testConfig);
+    }
+
```
Comment (umahaea) [inline src/test/java/compoundsTests/DropdownTest.java]: We need to have negative tests for the config validation. For eg. If I pass an incorrect element Id or incorrect component name.

Comment (umahaea) [inline src/test/java/compoundsTests/DropdownTest.java]: If possible, please group 1 or 2 tests as desktop-ci. Now we got so many in compounds.

File: src/main/java/compounds/jsfiles/inputs/inputs.js (hunk 2 above)
Comment (udhadpa) [inline src/main/java/compounds/jsfiles/inputs/inputs.js]: Oh good catch. Was checking something on inputs basic/fancy.. I added this line.
I will remove it

File: src/main/shell_scripts/components.sh (hunk 4 above)
Comment (udhadpa) [inline src/main/shell_scripts/components.sh]: Yes I know that.. But the icons, fonts and elements.css comes from DES-611 branch of elements. which is not merged and released. Inorder for the tests to pass I disabled it.

File: src/test/java/compoundsTests/DropdownTest.java (hunk 6 above)
Comment (udhadpa) [inline src/test/java/compoundsTests/DropdownTest.java]: The responsive behavior works only on smaller breakpoint, which is followed by iphone 6 and 7.
Rest of the devices have a different layout which is same like desktop. I have covered the behavior for both

File: src/main/shell_scripts/components.sh (hunk 4 above)
Comment (umahaea) [inline src/main/shell_scripts/components.sh]: Ok cool, got it

File: src/main/java/compounds/css/elements.css (hunk 1 above)
Comment (udhadpa) [inline src/main/java/compounds/css/elements.css]: It requires the elements.css from DES-611. If I removed it, the tests will fail.
Once the PR is approved. I will merge DES-611 in elements and release it. We can then remove the elements.css and enable the copy steps from the script file

File: src/main/java/compounds/jsfiles/inputs/inputs.js (hunk 2 above)
Comment (udhadpa) [inline src/main/java/compounds/jsfiles/inputs/inputs.js]: removed it

File: src/main/resources/environment.properties (hunk 3 above)
Comment (udhadpa) [inline src/main/resources/environment.properties]: done

**Comment by udhadpa** [comment]
Got approvals. Good to merge

**Pull request merged**